### DISCUSSION TITLE
Bluetooth pairing requests "in the middle of the night"

### DISCUSSION THREAD

**Discussion by emp-00** [Navid200/xDrip#171] [Dexcom]
Running xDrip+ 2021.09.02 with Dexcom G6 on a "Samsung A52s 5G" on Android 11 and experiencing Bluetooth pairing requests "all of a sudden". Seems like Android is losing the BT connection and the transmitter needs to be paired once again. Getting these alarms 2-3 times every week is very annoying. It always takes exactly 30 mins with "no readings" and then the BT pairing request appears. I "only" need to press "Pair" and then everything runs fine again. See screenshot with the logs and various error messages below. Why does this happen? Any configuration setting I need to tweak? 

FYI: I have the Android 11 smartphone since 4 weeks and before that I had an older Samsung phone with Android 10 - these issues clearly are related to the new smartphone!

![BT-error](https://user-images.githubusercontent.com/60714349/147562583-44500b98-243b-4562-bd74-5e51da292ca3.jpg)

**Comment by Navid200** [comment]
I had the exact same experience when I first got my current phone (Android 11).
I see similar experiences reported by others, on facebook, with new phones.

I have the following that may help.  
https://github.com/Navid200/xDrip/wiki/Intermittent-Connectivity

**Reply by emp-00** [reply]
Thanks for the link - I already used your recommended settings, but checking again now I found another hint => disabled battery optimization of the "Bluetooth" App as well. This is more or less the only thing that was different (Minimize Scanning & Avoid Scanning I have left ON for the time being). Will report back if this solved the problem - need to wait this out for a few days.

**Reply by emp-00** [reply]
In the meantime I have successfully updated to the latest nightly 2021.12.24. Settings changes see above. However: No success -> 36 hours after the changes I experienced another drop of the BT connection "in the middle of the night" :-( --- same procedure as always, need to confirm the Android pairing request and then it works again.

Today I disabled "Minimize Scanning" as the next testing step. Just heard that this might help for Samsung phones. Will keep you posted. Any other suggestions are welcome...

**Reply by Foxy7** [reply]
@emp-00  (Unclear how this will work with Android 11) but our main phone is Samsung S7 Android 8 and i use Blue Car Tethering so once TX is paired i tick the box in that App to force it to connect, i've been using it for ages now and not had any issues since.

The last link in the list on the OpenAPS docs: 
https://openaps.readthedocs.io/en/latest/docs/Customize-Iterate/bluetooth-tethering-edison.html#additional-app-requirement-on-android-to-enable-automatic-bt-tethering-reconnects

**Reply by emp-00** [reply]
Thank you @Foxy7 - very interesting! I just installed the "Blue Car Tethering" app. It's dated 2016 and throws a warning when started "developed for an older Android release" but it seems to run. I could select my Dexcom transmitter device and now I just need to wait and see? Does the app always keep running in the background even when closing it? Will keep you posted if this works. Just last night I got my latest "nightly alarm" ... Currently this happens approx. every 2nd night - and 90% of the time really at night, whatever that means. :-/

**Reply by Foxy7** [reply]
Good luck. I had issues with CamAPS keeping connected to the transmitter when we first started using it in September, so feel it's a Dexcom glitch rather than xDrip specific and Blue Car Tethering has worked for us there too.

Overall i've found bluetooth to be brilliant but fragile, we have a Bluejay GTS watch which is awesome standalone, but doesn't work with xDrip using an Android 8 phone, so currently they are separate. I only realised that after trying it on my Android 11 phone as others were having success and it seemed to be the only difference. Problem was there is no error message, just the bluetooth channels get blocked and i lost readings for both CamAPS and OpenAPS when using xDrip + Bluejay (as collector/phone collector turned off) on an Android 8 phone.

Long story short, I need a new phone..

**Reply by emp-00** [reply]
@Foxy7 : Does the "Blue Car Tethering" app always keep running in the background even when closing it?

**Reply by Foxy7** [reply]
@emp-00  I didn't know, so just checked and yes it does. Just had a look on my phone and shows as live via the "Running Services" menu, this is despite not having the app open.

was it any help?

**Reply by emp-00** [reply]
Thanks @Foxy7 - same here, it's visible under "Running Services" thus confirmed it's running. FYI: Since 1.5 days no more signal losses have occured - tonight or latest tomorrow is the decisive night ... fingers crossed.

**Reply by emp-00** [reply]
Since ~5 days after installing "Blue Car Tethering" I had **no more** BT connection dropouts and **no more** pairing requests! Problem fixed!!! :-) Thank you very much @Foxy7 : Looks like your App download suggestion works for me.

After all it's clearly an xDrip/Android10+11 issue. Would be great if this could be fixed in the xDrip code in the future. Unfortunately I don't have a hint where/how to fix this -> but the Tethering App functionality might offer an approach to find the bug or to find workaround in xDrip+.

**Reply by Foxy7** [reply]
@emp-00  that's good news, happy to hear you have a fix.

My initial feeling that it was a Dexcom issue, but this doesn't fit with the fact that the phone loses the pairing to the TX, so must be an Android glitch.

Note it's not an xdrip issue, as the same thing happened for me using CamAPS on a phone without xDrip (blue car tethering fixed this issue too).

@Navid200  perhaps on your intermittent connectivity page you can put a reference to the OpenAPS link above as a workaround?

**Reply by Navid200** [reply]
@Foxy7 I'm looking at that link and am not sure where to look to see a suggestion that may help intermittent connectivity.

**Reply by Foxy7** [reply]
@Navid200  it's the link under the yellow highlighter below. but guess if not clear for you then unlikely to help others. maybe just a straightforward suggestion to try a car bluetooth tethering app, e.g. "Blue Car Tethering", not sure on your policy of referencing 3rd party apps.

![image](https://user-images.githubusercontent.com/37312469/149189461-03168856-c915-4ae4-953b-f2ce085e91c6.png)

**Comment by dagord** [comment]
Sorry to jump in.
I've a similar issue on my Xiaomi Redmi Note 10 running Android 12: the bt connection with Dexcom transmitter drops after every reading and it is restored after 15 minutes; if I'm away more than 30 minutes I need to pair the device again.
This didn't happen on my old phone, a Xiaomi as well, running Android 10.
I've read other Dexcom users complaining after upgrading to Android 12 on different phone brands.
I partially solved installing BT auto connect: I get a new reading every 10 minutes and if I'm away the connection restores automatically when I come back.
But ideally I should get readings every 5 minutes! So I'm wondering if someone has found a working solution without having to use a third-part app.
